src/settings/Passwd.js

```diff
@@ -0,0 +1,198 @@
+  Button,
+  Callout,
+  Headline,
+  Pane,
+  TextField
+} from '@folio/stripes/components';
+
+/**
+ * Reset a user's password
+ */
+class Passwd extends React.Component {
+  static manifest = Object.freeze({
+    passwd: {
+      type: 'okapi',
+      path: 'authn/credentials',
+      fetch: false,
+      accumulate: true,
+    },
+    userExists: {
+      type: 'okapi',
+      path: 'users',
+      fetch: false,
+      accumulate: true
+    },
+    isLocalPasswordSet: {
+      type: 'okapi',
+      path: 'authn/credentials-existence',
+      fetch: false,
+      accumulate: true,
+    },
+    moduleId: '',
+  });
+
+  static propTypes = {
+    stripes: PropTypes.shape({
+      connect: PropTypes.func.isRequired,
+    }).isRequired,
+    resources: PropTypes.shape({
+      passwd: PropTypes.object,
+      isLocalPasswordSet: PropTypes.object,
+    }).isRequired,
+    mutator: PropTypes.shape({
+      passwd: PropTypes.object,
+    }),
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.callout = React.createRef();
+  }
+
+  validate = (values) => {
+    const errors = {};
+
+    if (!values.username) {
+      errors.username = <FormattedMessage id="stripes-core.label.missingRequiredField" />;
+    }
+    if (!values.password) {
+      errors.password = <FormattedMessage id="stripes-core.label.missingRequiredField" />;
+    }
+    if (!values.confirm) {
+      errors.confirm = <FormattedMessage id="stripes-core.label.missingRequiredField" />;
+    } else if (values.confirm !== values.password) {
+      errors.confirm = <FormattedMessage id="ui-developer.passwd.error.confirm" />;
+    }
+    return errors;
+  };
+
+  handleSubmit = (values, form) => {
+    const { mutator, stripes, intl } = this.props;
+    let userId;
+
+    // GET the user by username in order to get their UUID.
+    // .then(check whether that UUID already has credentials assigned)
+    // .then(if yes, delete them then POST the new creds; if no, POST the new creds)
+    // .then(
+    //    callout about it, callout everything you feel
+    //    give that UI notice that this password is for real
+    // ).catch(
+    //    callout about it, callout even if you bomb
+    //    give that UI warning that something has gone so wrong
+    // )
+    mutator.userExists.GET({ params: { query: `username==${values.username}` } })
+      .then((res) => {
+        userId = res.users[0]?.id;
+        if (!userId) {
+          throw new Error(intl.formatMessage({ id: 'ui-developer.passwd.error.missingUser' }, { username: values.username }));
+        }
+      })
+      .then(() => mutator.isLocalPasswordSet.GET({ params: { userId } }))
+      .then(res => {
+        const credentials = {
+          username: values.username,
+          password: values.password,
+          userId,
+        };
+
+        if (!res.credentialsExist) {
+          return mutator.passwd.POST(credentials);
+        } else {
+          // raw fetch because stripes-connect DELETE mutators can't handle
```

Comment:
Oh, what a rancid abstraction-leak. I agree, though, that there doesn't seem to be anything very obvious you can do about it.

Unless ...
```
  static manifest = Object.freeze({
    // ...
    changePasswd: {
      type: 'okapi',
      path: 'authn/credentials',
      fetch: false,
      params: {
        userId: '%{id}',
      }
    },
};

// ...

mutator.changePasswd.DELETE({ userId })

Reply:
On reflection, I think using a DELETE mutator here may be a dead end: I ran into similar issues in Course Reserves and ended up doing [this](https://github.com/folio-org/ui-courses/blob/master/src/components/ViewCourse/sections/ViewCourseReserves.js#L71-L74) and [this](https://github.com/folio-org/ui-courses/blob/master/src/components/ViewCourse/sections/ViewCourseInstructors.js#L16-L19), which suggests I tried the mutator route and couldn't make it work. I have only vague memories of this, but I seem to recall that stripes-connect doesn't provide a suitable way to inject the `userId` value into the template.

So stick with your use of `fetch`. (BTW., `ky` which I use in Course Reserves is just a simplifying layer over `fetch` which doesn't require quite so much badgering about with exceptions. I prefer it, but since you have the raw-`fetch` version, there is no reason to change.)